azure/converters/diagnostics.go

```diff
@@ -17,11 +17,41 @@ limitations under the License.
 package converters
 
 import (
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v5"
 	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2021-11-01/compute"
 	"k8s.io/utils/ptr"
 	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1beta1"
 )
 
+// GetDiagnosticsProfileV2 converts a CAPZ Diagnostics option to a Azure SDK Diagnostics Profile.
+func GetDiagnosticsProfileV2(diagnostics *infrav1.Diagnostics) *armcompute.DiagnosticsProfile {
```

Comment:
To point out one example of this "clone until everything is converted" strategy, `GetDiagnosticsProfile` is called by both the scalesets and virtualmachines services. We're converting scalesets, but don't want to convert virtualmachines yet.

So we clone the function, add a `V2` suffix, and change it to use `armcompute` types instead of old `compute` types. We also clone its unit tests.

Then when both scalesets and virtualmachines have converted to SDK v2, we get rid of the old SDK v1-based function that's no longer used, and we rename the -`V2` function back to the original name.